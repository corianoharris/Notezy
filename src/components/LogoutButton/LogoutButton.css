.logout-button-container {
  width: 100%;
  margin-bottom: 20px;
  text-align: right;
}



@media screen and (min-width: 1024px) {
  .logout-button-container {
    width: auto;
    margin-bottom: none;
  }
}