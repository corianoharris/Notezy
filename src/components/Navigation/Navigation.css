.navbar {
  padding-bottom: 30px;
  width: 100vw;
  height: 200px;
  background-color: #3c005a;
}

.navbar-nav {
  align-items: center;
  gap: 10px;
}

.navbar-brand {
  font-size: 1rem;
}

.nav-link, .nav-link:visited, .nav-link:link {
  padding-bottom: 5px;
  width: auto;
  color: darkgrey;
}

.nav-link:hover {
  font-weight: 700;
  padding-bottom: 5px;
  color: white;
  border-bottom: 1px solid white;
}

.nav-link:active  {
  color: white;
  font-weight: 700;
}

.active {
  font-weight: bolder;
  color: grey;
}


.greeting-container {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  top: 10px;
  right: 10px;
  width: auto;
}

.brand-link {
  display: flex;
  align-items: center;
  gap: 5px;
}

.brand-name {
  font-size: 1.725rem;
  pointer-events: none;
  cursor: default;
}

.branding-image {
  margin-left: 10px;
  width: 16px;
  height: 16px;
}

@media screen and (min-width: 480px) {
  .navbar-brand {
    font-size: 1.5rem;
  }
}

@media screen and (min-width: 990px) {
  .branding-image {
    height: 32px;
    width: 32px;
  }
}

@media screen and (max-width: 900px) {
  .navbar-nav {
    display: block;
  }
}