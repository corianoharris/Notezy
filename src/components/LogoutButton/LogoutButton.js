import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import "./LogoutButton.css";
import Loader from "../Loader";

const LogoutButton = () => {
	const { logout, isLoading } = useAuth0();
	if (isLoading) {
		return <Loader />;
	}

	return (
		<div className='logout-button-container'>
		<button
			className="btn btn-outline-light"
			onClick={() => logout({ returnTo: window.location.origin })}>
			Log Out
			</button>
			</div>
	);
};

export default LogoutButton;