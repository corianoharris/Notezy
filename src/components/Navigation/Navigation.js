import React from 'react';
import LogoutButton from '../LogoutButton';
import './Navigation.css';

const  Navigation = () => {
	return (
		<nav className="navbar navbar-dark bg-dark navbar-expand-lg border-bottom-3 shadow ">
			<a
				href={'/'}
				className="navbar-brand links"
				style={{
					fontFamily: 'Permanent Marker',
					fontSize: '48px',
					color: 'white'
				}}>
				Notezy
			</a>
			<div className="collpase navbar-collapse">
				<ul className="navbar-nav mr-auto">
					<li className="navbar-item">
						<a
							href={'/note'}
							className="nav-link active links"
							style={{
								color: '#d3d3d3d3',
							}}>
							Create a note
						</a>
					</li>
					<li className="navbar-item">
						<a
							href={'/notes'}
							className="nav-link links"
							style={{
								color: '#d3d3d3d3',
							}}>
							Your Notes
						</a>
					</li>
				</ul>
			</div>
			<LogoutButton />
		</nav>
	);
};

export default Navigation;
