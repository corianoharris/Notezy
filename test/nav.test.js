import React from 'react';
import Navigation from '../src/components/Navigation';
import LogoutButton from '../src/components/LogoutButton';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// react test renderer to test ui
it('renders correctly', () => {
	const nav = renderer.create(< Navigation />).toJSON();
	const logout = renderer.create(<LogoutButton />).toJSON();
	expect(nav).toMatchSnapshot();
	expect(logout).toMatchSnapshot();
});

// testing-library/react ti test nav links
test('renders nav links', () => {
	const { getByText } = render(< Navigation />);
	const NotezyLinkText = getByText(
		/Notezy/i,
	);
	const CreateANoteLinkText = getByText(
		/Create a note/i,
	);
	const YourNotesLInkText = getByText(

		/Your notes/i,
	);

	expect(NotezyLinkText).toBeInTheDocument();
	expect(CreateANoteLinkText).toBeInTheDocument();
	expect(YourNotesLInkText).toBeInTheDocument();
});
