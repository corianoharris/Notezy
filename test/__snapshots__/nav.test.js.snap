// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-dark bg-dark navbar-expand-lg border-bottom-3 shadow "
>
  <div
    className="collpase navbar-collapse"
  >
    <ul
      className="navbar-nav mr-auto mt-2"
    >
      <li
        className="navbar-item"
      >
        <div
          className="brand-link"
        >
          <img
            alt="notebook icon"
            className="branding-image"
            src="test-file-stub"
          />
           
           
          <a
            className="navbar-brand link-underline-light link-underline-opacity-100-hover"
            href="/"
          >
            Notezy
          </a>
        </div>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="nav-link active link-underline-light link-underline-opacity-100-hover"
          href="/createanote"
        >
           Create a note
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="nav-link link-underline-light link-underline-opacity-100-hover"
          href="/yournotes"
        >
          Your notes
        </a>
      </li>
    </ul>
  </div>
  <div
    className="greeting-container"
  >
    <div
      className="Loader"
    >
      Loading your greeting...
    </div>
    <div
      className="text-center p-5 mx-auto border-0 login"
    >
      <h2
        className="container w-100 mt-5 mx-auto text-center "
      >
        Getting your page...
      </h2>
    </div>
    <div
      className="footer text-dark mt-2"
    >
      <h6>
        authored by
         
        <span>
          <a
            className="footer-link badge badge-light"
            href="https://www.corianoharris.com"
          >
            Coriano Harris
          </a>
        </span>
      </h6>
    </div>
  </div>
</nav>
`;

exports[`renders correctly 2`] = `
Array [
  <div
    className="text-center p-5 mx-auto border-0 login"
  >
    <h2
      className="container w-100 mt-5 mx-auto text-center "
    >
      Getting your page...
    </h2>
  </div>,
  <div
    className="footer text-dark mt-2"
  >
    <h6>
      authored by
       
      <span>
        <a
          className="footer-link badge badge-light"
          href="https://www.corianoharris.com"
        >
          Coriano Harris
        </a>
      </span>
    </h6>
  </div>,
]
`;
